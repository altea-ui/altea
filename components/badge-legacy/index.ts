import BadgeLegacy from './badge-legacy'

export type { BadgeLegacyProps } from './badge-legacy'
export default BadgeLegacy
