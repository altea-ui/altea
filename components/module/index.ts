import Module from './module'

export type { ModuleProps } from './module'
export default Module
