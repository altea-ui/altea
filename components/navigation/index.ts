import NavigationItem from './item';
import NavigationSubItem from './sub-item';
import Navigation from './navigation';

export interface INavigationItem {
  title?: string;
  url?: string;
  desc?: string;
  icon?: React.ReactNode;
  exactMatch?: boolean;
}

export type NavigationItemComponentType = typeof NavigationItem & {
  Child: typeof NavigationSubItem;
};
(NavigationItem as NavigationItemComponentType).Child = NavigationSubItem;

export type NavigationComponentType = typeof Navigation & {
  Item: NavigationItemComponentType;
};
(Navigation as NavigationComponentType).Item = NavigationItem as NavigationItemComponentType;

export type { NavigationItemProps } from './item';
export default Navigation as NavigationComponentType;
