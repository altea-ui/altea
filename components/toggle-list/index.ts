import ToggleList from './toggle-list'
import ToggleItem from './toggle-list-item'

export * from './shared'
export type ToggleComponentType = typeof ToggleList & {
  Item: typeof ToggleItem
}
;(ToggleList as ToggleComponentType).Item = ToggleItem
export default ToggleList as ToggleComponentType
