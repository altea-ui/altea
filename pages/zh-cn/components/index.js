import redirect from 'lib/redirect'

export default redirect('/zh-cn/components/text')
