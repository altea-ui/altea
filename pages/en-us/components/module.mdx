import { Layout, Playground, Attributes } from 'lib/components'
import { Module } from 'components'

export const meta = {
  title: 'Modal',
  group: 'Feedback',
}

## Module

The Module component offers a structured panel for data display, featuring an optional label and footer. It's ideal for clean, organized information presentation.

<Playground
  title="Basic"
  desc="A basic module with a title and content."
  scope={{ Module }}
  code={`
() => {
  return (
    <Module name="Title">
        This module details something important.
    </Module>
  )
}
`}
/>
