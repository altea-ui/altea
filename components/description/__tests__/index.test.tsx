import React from 'react'
import { mount } from 'enzyme'
import { Description } from 'components'

describe('Description', () => {
  it('should render correctly', () => {
    const wrapper = mount(<Description title="title" />)
    expect(wrapper).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should render react-node on title', () => {
    let wrapper = mount(<Description title={<p>p</p>} />)
    expect(wrapper.find('p').length).not.toBe(0)

    wrapper = mount(<Description title={<h1>p</h1>} />)
    expect(wrapper.find('h1').length).not.toBe(0)
  })

  it('should render react-node on content', () => {
    let wrapper = mount(<Description content={<p>p</p>} />)
    expect(wrapper.find('p').length).not.toBe(0)

    wrapper = mount(<Description content={<h1>p</h1>} />)
    expect(wrapper.find('h1').length).not.toBe(0)
  })
})
