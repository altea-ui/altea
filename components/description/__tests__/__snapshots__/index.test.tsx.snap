// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description should render correctly 1`] = `ReactWrapper {}`;
